src/ui/svgWidget.js

```diff
@@ -0,0 +1,78 @@
+//////////////////////////////////////////////////////////////////////////////
+/**
+ * Create a new instance of class geo.gui.svgWidget
+ *
+ * Due to the nature of d3 creating DOM elements as it inserts them, calls to appendChild
+ * don't appear in this widget.
+ *
+ * The canvas of an svgWidget always refers to the actual svg element.
+ * The parentCanvas can refer to another widgets svg element, dom element, or the
+ * UI layers dom element.
+ * See {@link geo.gui.widget#parentCanvas}.
+ *
+ * @class
+ * @extends geo.gui.domWidget
+ * @returns {geo.gui.svgWidget}
+ *
+ */
+//////////////////////////////////////////////////////////////////////////////
+geo.gui.svgWidget = function (arg) {
+  'use strict';
+  if (!(this instanceof geo.gui.svgWidget)) {
+    return new geo.gui.svgWidget(arg);
+  }
+
+  geo.gui.domWidget.call(this, arg);
+
+  var m_this = this,
+      m_renderer = null;
+
+  this._init = function (arg) {
+    var d3Parent;
+    arg = m_this.parseArgs(arg);
+    m_this.args = arg;
```

Comment:
The ```svgWidget``` constructor is [different enough](https://github.com/OpenGeoscience/geojs/blob/2b72de6a4162ebd392b0bdf86896cbcb5c87eebf/src/ui/svgWidget.js#L35-L42) that I didn't want it to call the superclass ```_init```. The only really common operation among all of the constructors is the argument parsing. It could probably be done more cleanly by actually setting them on ```m_this``` - but if possible, I would prefer to keep the svg/dom constructors separate.